src/main/java/uk/gov/companieshouse/search/api/exception/EndpointException.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.companieshouse.search.api.exception;
+
+import java.net.MalformedURLException;
+
+/**
+ * EndpointException is a wrapper exception that hides
+ * lower level exceptions from the caller and prevents them
+ * from being propagated up the call stack.
+ */
+public class EndpointException extends MalformedURLException {
```

Comment:
It would be better to extend `RuntimeException` here as the use-case where this exception is thrown is unrecoverable (i.e. the app is misconfigured and will not function correctly) and we ideally want the application to terminate at this point with a sensible error. It's also worth updating the Javadoc comment here to represent the use of this class in that case.